### ISSUE TITLE
Line breaks ?

### ISSUE THREAD

**Issue by mdnava** [JanDeDobbeleer/oh-my-posh#1837]
### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

I've used for months now the same Posh config file for Cmd, Pwsh, Msys2 Bash, Git Bash, Fish and Zsh. But after I updated from v6.43.0 to v7.27.0 several issues popped that I was able to fix, but an important \n line break stopped working:

```
{
  "type": "text",
  "style": "plain",
  "properties": {
	"prefix": "",
	"postfix": "",
	"text": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
  }
},
```
How it should be (v6.43.0) vs. how it currently is (v7.27.0):

![2022-02-28 09 23 16](https://user-images.githubusercontent.com/24944618/155991138-e5351624-c942-448c-8183-7bb507279039.png) ![2022-02-28 09 25 21](https://user-images.githubusercontent.com/24944618/155991140-f9fdfca7-0257-428b-acba-6efe0893875a.png)

It seems the \n is no longer recognized.. Is this a bug? an intended change?..
If so, how can I conditionally add a line break ?

The variable $PROMPT_PRE_LNBR is pre-set on every shell when is needed, so for example, a line break is added only to Pwsh and Bash, but not to Cmd. I can't use the "newline" parameter because it would then apply to every shell.

This prompt had been working great in every shell for months now, since I found out about OhMyPosh; but it broke after updating to v7.27.0 today. I did manage to fix most issues (mostly spacing and variables) but this line break issue remains.

Can anyone comment?

### Theme

No theme. My own custom prompt.

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

other (please specify)

### Log output

```shell
Version: 7.27.0

Segments:

ConsoleTitle(false) -   0 ms -
text(false) -   1 ms -

text(true)  -   1 ms -  PS7
path(true)  -   0 ms -  \Desktop 
time(true)  -   1 ms -  07:00:25 
text(false) -   0 ms -
git(false)  -  17 ms -
text(true)  -   0 ms - ϟ

Run duration: 21.0007ms

Cache path: C:\Users\USERNAME\AppData\Local/oh-my-posh

Logs:

2022/02/28 07:00:25 Args duration: 0s, args:
2022/02/28 07:00:25 Args duration: 0s, args:
2022/02/28 07:00:25 Args duration: 0s, args:
2022/02/28 07:00:25 Shell duration: 20.7267ms, args:
2022/02/28 07:00:25 debug: Getenv

2022/02/28 07:00:25 Getenv duration: 0s, args: OMP_CACHE_DISABLED
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 Root duration: 0s, args:
2022/02/28 07:00:25 Shell duration: 0s, args:
2022/02/28 07:00:25 ErrorCode duration: 0s, args:
2022/02/28 07:00:25 IsWsl duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 User duration: 0s, args:
2022/02/28 07:00:25 Host duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 Args duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 PathSeparator duration: 0s, args:
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 IsWsl duration: 0s, args:
2022/02/28 07:00:25 StackCount duration: 0s, args:
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 GOOS duration: 0s, args:
2022/02/28 07:00:25 HasCommand duration: 17.9995ms, args: git.exe
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 error: HasParentFilePath
CreateFile D:\.git: The system cannot find the file specified.
2022/02/28 07:00:25 HasParentFilePath duration: 0s, args: .git
2022/02/28 07:00:25 debug: Pwd
D:\USERNAME\Desktop
2022/02/28 07:00:25 Pwd duration: 0s, args:
2022/02/28 07:00:25 debug: Home
D:\USERNAME
2022/02/28 07:00:25 TemplateCache duration: 0s, args:
2022/02/28 07:00:25 debug: Getenv
C:\Users\USERNAME\AppData\Local
2022/02/28 07:00:25 Getenv duration: 0s, args: LOCALAPPDATA
2022/02/28 07:00:25 CachePath duration: 0s, args:
```

**Comment by JanDeDobbeleer** [comment]
@mdnava that should be 

```json
{
  "type": "text",
  "style": "plain",
  "properties": {
	"template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
  }
},
```

Weird that this wasn't migrated correctly. Because we do.

**Comment by mdnava** [comment]
> @mdnava that should be
> 
> ```json
> {
>   "type": "text",
>   "style": "plain",
>   "properties": {
> 	"template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
>   }
> },
> ```
> 
> Weird that this wasn't migrated correctly. Because we do.

Isn't it the same code? I copied/pasted and it still doesn't work.

I've already tried it with the \n alone and it seems the line break is no longer recognized.

**Comment by JanDeDobbeleer** [comment]
@mdnava the `text` property isn't used anymore. That one moved to `template` We do print line breaks ourselves so they should work but I'll check.

**Comment by mdnava** [comment]
Removing the "text" is somewhat confusing, but I have to mention that I have several other "type: text" segments and all seem to be working fine. The only problem I see is with the \n line break. If I put any other character (like , . -) it's shown in the prompt.

**Comment by JanDeDobbeleer** [comment]
@mdnava wait, I know where this comes from. We did a change that doesn't enable a segment when it's whitespace and a line break is white space. I'll fix it so it does this only for texts of length 0.

**Comment by JanDeDobbeleer** [comment]
> Removing the "text" is somewhat confusing

@mdnava that's because we added templates to all segments and keep consistency as to where you add "text".

**Comment by mdnava** [comment]
> @mdnava wait, I know where this comes from. We did a change that doesn't enable a segment when it's whitespace and a line break is white space. I'll fix it so it does this only for texts of length 0.

Yep, you're right. I added a dot after the \n and the line break now works. It's because it's considered a whitespace. But then now there's a leftover character.

> @mdnava that's because we added templates to all segments and keep consistency as to where you add "text".

Got it!.. I'm gonna read again the documentation..

**Comment by inakiabt** [comment]
```json
{
          "type": "text",
          "style": "plain",
          "properties": {
            "template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
          }
        },
```

Doesn't seem to work to me :(
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/606852/185006929-dcd0cfa6-5a32-43a6-bd80-7a1c5e54c0ff.png">
No extra line between commands is added.

My config:
```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "type": "text",
          "style": "plain",
          "properties": {
            "template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
          }
        },
        {
          "background": "#003543",
          "foreground": "#00c983",
          "leading_diamond": "\ue0b6",
          "style": "diamond",
          "template": "{{ .Icon }} ",
          "type": "os"
        },
        {
          "background": "#468488",
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "folder_separator_icon": "/",
            "home_icon": "~",
            "style": "full"
          },
          "style": "powerline",
          "template": " \uf74a  {{ .Path }} ",
          "type": "path"
        },
        {
          "type": "command",
          "style": "powerline",
          "leading_diamond": "\ue0b6",
          "powerline_symbol": "\ue0b0",
          "trailing_diamond": "\ue0b4",
          "background": "#a45226",
          "foreground": "#ffffff",
          "properties": {
            "shell": "zsh",
            "command": "git remote get-url origin"
          },
          "template": " \uF1D3 {{ .Output }} "
        },
        {
          "background": "#FCA17D",
          "foreground": "#ffffff",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#FF9248{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#ff4500{{ end }}",
            "{{ if gt .Ahead 0 }}#B388FF{{ end }}",
            "{{ if gt .Behind 0 }}#B388FF{{ end }}"
          ],
          "leading_diamond": "\ue0b6",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "branch_icon": "<#ffffff>\ue0a0 </>",
            "branch_max_length": 25,
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          },
          "style": "powerline",
          "template": " {{ .HEAD }}{{ .BranchStatus }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \uf692 {{ .StashCount }}{{ end }} ",
          "trailing_diamond": "\ue0b4",
          "type": "git"
        },
        {
          "background": "#76b367",
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "style": "powerline",
          "template": " \ue718 {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }} ",
          "type": "node"
        },
        {
          "background": "#2e9599",
          "background_templates": [
            "{{ if gt .Code 0 }}red{{ end }}"
          ],
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "always_enabled": true
          },
          "style": "diamond",
          "template": " {{ if gt .Code 0 }}\uf525{{ else }}\uf469{{ end }}",
          "trailing_diamond": "\ue0b4",
          "type": "exit"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "right",
      "segments": [
        {
          "background": "#0077c2",
          "foreground": "#ffffff",
          "style": "plain",
          "template": "<#0077c2,transparent>\ue0b6</> \uf489 {{ .Name }} <transparent,#0077c2>\ue0b2</>",
          "type": "shell"
        },
        {
          "background": "#f36943",
          "background_templates": [
            "{{if eq \"Charging\" .State.String}}#40c4ff{{end}}",
            "{{if eq \"Discharging\" .State.String}}#ff5722{{end}}",
            "{{if eq \"Full\" .State.String}}#4caf50{{end}}"
          ],
          "foreground": "#ffffff",
          "invert_powerline": true,
          "powerline_symbol": "\ue0b2",
          "properties": {
            "charged_icon": "\ue22f ",
            "charging_icon": "\ue234 ",
            "discharging_icon": "\ue231 "
          },
          "style": "powerline",
          "template": " {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}\uf295 ",
          "type": "battery"
        },
        {
          "background": "#83769c",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "leading_diamond": "\ue0b2",
          "properties": {
            "always_enabled": true
          },
          "style": "diamond",
          "template": " \uf608 {{ .FormattedMs }} <#111111>\ue0b2</>",
          "type": "executiontime"
        },
        {
          "background": "#33658A",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "leading_diamond": "\ue0b2",
          "properties": {
            "time_format": "3:04 PM"
          },
          "style": "diamond",
          "template": " \uf5ef {{ .CurrentDate | date .Format }}",
          "trailing_diamond": "\ue0b4",
          "type": "time"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "foreground": "#cd5e42",
          "style": "plain",
          "template": "\ue3bf ",
          "type": "root"
        },
        {
          "foreground": "#CD4277",
          "style": "plain",
          "template": " <#45F1C2><b>\u26a1</b></><b>{{ .UserName }}</b> <#26C6DA>\u276f</><#45F1C2>\u276f</>",
          "type": "text"
        },
        {
          "type": "iterm",
          "style": "plain",
          "foreground": "#80ffea",
          "template": "{{ .PromptMark }}"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "{{ .Shell }} in {{ .Folder }}",
  "final_space": true,
  "version": 2
}
```

**Comment by JanDeDobbeleer** [comment]
@inakiabt this all depends on `PROMPT_PRE_LNBR` being set. Otherwise that segment will not output anything. That and being on the latest oh-my-posh.

**Comment by inakiabt** [comment]
@JanDeDobbeleer Found the issue:

It should be:
```json
{
  "type": "text",
  "style": "plain",
  "template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
}
```

instead of:
```json
{
  "type": "text",
  "style": "plain",
  "properties": {
    "template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}"
  }
}
```

**Comment by JanDeDobbeleer** [comment]
@inakiabt absolutely right, not sure why I missed that 🤦🏻

**Comment by mdnava** [comment]
@JanDeDobbeleer The same prompt I'm using in this issue (with given fixes) stopped working in Fish since I updated Msys2 Fish to its latest version a few weeks ago. Before that it was working fine in Fish, and still works fine in Cmd/Pwsh/Bash/Zsh.

OhMyPosh is up to date. I'm not particularly using much the Fish shell. Should I create a new issue for this? (not a Posh issue?)

This is the setting for the Fish prompt (I've also tried using absolute paths):
```
set -gx PROMPT_PRE_LNBR "TRUE"
"$Utilities/OhMyPosh/oh-my-posh.exe" --init --shell fish --config "$EnvScripts/AutoRun/posh-prompt.json" | source
```

This is the error I get:

```
fish: Unknown command: /d/mdnava/Desktop/D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe- (line 32):
    D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe print primary --config $POSH_THEME --shell fish --error $omp_status_cache --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION
    ^
in function 'fish_prompt'
in command substitution
fish: Unknown command: /d/mdnava/Desktop/D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe- (line 49):
    D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe print right --config $POSH_THEME --shell fish --error $omp_status_cache --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION
    ^
in function 'fish_right_prompt'
in command substitution
```


I've also tried using absolute paths absolute paths in the prompt setting:

```
set -gx PROMPT_PRE_LNBR "TRUE"
D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe --init --shell fish --config "D:/mdnava/OneDrive/Command/EnvScripts/AutoRun/posh-prompt.json" | source
```
The error I get with absolute paths:
```
fish: Unknown command: /d/mdnava/Desktop/D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe~/.config/fish/config.fish (line 7):
D:/mdnava/OneDrive/Command/Utilities/OhMyPosh/oh-my-posh.exe --init --shell fish --config "D:/mdnava/OneDrive/Command/EnvScripts/AutoRun/posh-prompt.json" | source
^
from sourcing file ~/.config/fish/config.fish
        called during startup
```

**Comment by JanDeDobbeleer** [comment]
@mdnava not sure what exactly causes it other than msys2 being at the origin of it. What you could try is tell oh-my-posh to use the executable name, and not resolve the full path.

```
oh-my-posh init fish --strict --config  "$EnvScripts/AutoRun/posh-prompt.json" | source
```

**Comment by mdnava** [comment]
@JanDeDobbeleer Ok. Thank you!. Any of these commands work:

```
oh-my-posh.exe init fish --strict --config "$EnvScripts/AutoRun/posh-prompt.json" | source
```
```
"$Utilities/OhMyPosh/oh-my-posh.exe" init fish --strict --config "$EnvScripts/AutoRun/posh-prompt.json" | source
```
So it's "init" instead of "--init". 
However, both commands fail without the --strict flag.. which I can't seem to find in the help.

Also. the path in the prompt gets double backslashes (only in Fish — Other shells get a normal single backslash). It doesn't bother me at all. Just thought is was worth mentioning. 

![2022-08-21 03 55 50](https://user-images.githubusercontent.com/24944618/185781498-874a9254-fda2-40bb-89ec-c3a05b5fc809.png) ![2022-08-21 03 58 20](https://user-images.githubusercontent.com/24944618/185781554-d419989c-2526-40c9-8782-d714e9cbc619.png)

Fish version version 3.5.0 (June 16, 2022)

Have a nice weekend!

**Comment by JanDeDobbeleer** [comment]
@mdnava that's also because of msys. Remember that according to the executable we're running inside Windows, yet the shell emulates unix. We need to handle this case differently maybe, but it touches everything as the impact is tremendous.

**Comment by mdnava** [comment]
> @mdnava that's also because of msys. Remember that according to the executable we're running inside Windows, yet the shell emulates unix. We need to handle this case differently maybe, but it touches everything as the impact is tremendous.

In case is relevant, I also use Bash and Zsh from Msys2 (single backslashes):

![2022-08-21 05 08 00](https://user-images.githubusercontent.com/24944618/185783992-baf9d72b-9df7-4a89-9cf1-5c2f8cebc64b.png) ![2022-08-21 05 08 27](https://user-images.githubusercontent.com/24944618/185783993-516a2701-62f5-4808-9f1e-b3110b43c0ea.png)

And a test on Bash (Ubuntu – WSL2):

![2022-08-21 05 10 00](https://user-images.githubusercontent.com/24944618/185784147-5f72ff79-f830-40d4-a67f-106b37da5c98.png)

**Note:** All shells use the same 'posh-prompt.json' file.

What seems odd, is that the Fish prompt command issue, and the double backslash path, in both cases they happened after the latest Fish update. Before that it was working as intended. Curious.

**Comment by JanDeDobbeleer** [comment]
@mdnava I'll have a look once I'm home.

**Comment by JanDeDobbeleer** [comment]
@mdnava I can't find anything specific, but as 3.5 isn't the latest I'll point to fish (and msys) being the culprit here.

**Comment by mdnava** [comment]
> I can't find anything specific, but as 3.5 isn't the latest I'll point to fish (and msys) being the culprit here.

@JanDeDobbeleer I understand.. thank you!.

1. The prompt works fine in Fish with the `--strict` parameter (but not without it),
2. The double backslashes are gone as of v8.36.0 (latest version of Oh-My-Posh / 2022-08-28)

![2022-08-28 10 08 20](https://user-images.githubusercontent.com/24944618/187078330-86ad0f29-7d32-4a88-a8a9-2861bb444704.png)

**Comment by lewis-yeung** [comment]
@mdnava

> 1. The prompt works fine in Fish with the `--strict` parameter (but not without it),

It seems that MSYS fish cannot recognize a Windows-style path as MSYS bash does, so the command will fail without a `--strict` flag. 😆

> 2. The double backslashes are gone as of v8.36.0 (latest version of Oh-My-Posh / 2022-08-28)

I think this was fixed since [v8.35.2](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v8.35.2) (via #2716). :)
